pkgs/development/libraries/SDL/setup-hook.sh

```diff
@@ -1,9 +1,15 @@
 addSDLPath () {
   if [ -e "$1/include/SDL" ]; then
     export SDL_PATH="${SDL_PATH-}${SDL_PATH:+ }$1/include/SDL"
-  fi
-  if [ -e "$1/lib" ]; then
-    export SDL_LIB_PATH="${SDL_LIB_PATH-}${SDL_LIB_PATH:+ }-L$1/lib"
+    # NB this doesn’t work with split dev packages because different packages
+    # will contain "include/SDL/" and "lib/" directories.
+    #
+    # However the SDL_LIB_PATH is consumed by SDL itself and serves to locate
+    # libraries like SDL2_mixer, SDL2_image, etc which are not split-package
```

Comment:
`sdl2-mixer` has `out` and `dev` however `sdl-mixer` doesn't

and there's no `sdl-*` which have split outputs `fd --type d 'sdl_' pkgs/development/libraries | xargs -I {} cat "{}/default.nix" | grep outputs`

https://github.com/NixOS/nixpkgs/pull/296919

Reply:
maybe the lib file names could be matched for `sdl`?

Reply:
I think we're accidentally fine - `SDL_PATH` is only used by the `SDL` package. For `SDL2` there's `SDL2_PATH` and different setup hook, still employing similar structure and probably susceptible to `sdl2-mixer` having split outputs: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/SDL2/setup-hook.sh.

> Maybe the lib file names could be matched for sdl? that could work with split outputs

Do you mean that shell in `setup-hook.sh` would somehow analyse its argument to determine whether we're looking at an sdl components package?